### PR TITLE
Tomer/enable compressionable data

### PR DESCRIPTION
**Pull request by tomerr-lb**

### CODE REVIEW

File: memtier_benchmark.cpp [hunk 1]
```diff
@@ -181,22 +183,23 @@ static void config_print_to_json(json_handler * jsonhandler, struct benchmark_co
...
-    jsonhandler->write_obj("data_size_pattern" ,"\"%s\"", 		cfg->data_size_pattern);
+    jsonhandler->write_obj("data_size_range"   ,"\"%u:%u\"",   	cfg->data_size_range.min, cfg->data_size_range.max);
+    jsonhandler->write_obj("data_size_list"    ,"\"%s\"",      	cfg->data_size_list.print(tmpbuf, sizeof(tmpbuf)-1));
+    jsonhandler->write_obj("data_size_pattern" ,"\"%s\"",      	cfg->data_size_pattern);
     jsonhandler->write_obj("expiry_range"      ,"\"%u:%u\"",   	cfg->expiry_range.min, cfg->expiry_range.max);
     jsonhandler->write_obj("data_import"       ,"\"%s\"",       cfg->data_import);
     jsonhandler->write_obj("data_verify"       ,"\"%s\"",       cfg->data_verify ? "true" : "false");
     jsonhandler->write_obj("verify_only"       ,"\"%s\"",       cfg->verify_only ? "true" : "false");
-    jsonhandler->write_obj("generate_keys"     ,"\"%s\"",     	cfg->generate_keys ? "true" : "false");
+    jsonhandler->write_obj("generate_keys"     ,"\"%s\"",      	cfg->generate_keys ? "true" : "false");
```
Comment (yael-lb) [inline memtier_benchmark.cpp:202]: Generally, it's better to add a commit with commit title "cosmetics" that includes all the cosmetic changes to the code. It separates the actual implementation from other changes in the code and makes it easier to understand what you did.

File: obj_gen.cpp [hunk 2]
```diff
@@ -221,12 +223,18 @@ void object_generator::alloc_value_buffer(void)
         size = m_data_size.size_list->largest();
     }
 
-    m_value_buffer_size = size;
+    if (!m_compress_perc)
+        m_value_buffer_size = size;
+    else {
+        m_value_buffer_size = (float)size*(1-(float)m_compress_perc/100);
```
Comment (yael-lb) [inline obj_gen.cpp]: You're casting to float and then assigning into an unsigned int. Why casting the result to float and why not using static cast to do so?
I think the result should be statically cast to unsigned int.

File: memtier_benchmark.h [hunk 3]
```diff
@@ -50,6 +50,7 @@ struct benchmark_config {
     unsigned int clients;
     unsigned int threads;
     bool cpu_split;
+    int compress_perc;
```
Comment (yael-lb) [inline memtier_benchmark.h]: Consider using unsigned int, as the compress_perc value can't be negative

File: obj_gen.cpp [hunk 4]
```diff
@@ -430,7 +450,13 @@ data_object* object_generator::get_object(int iter)
 
     // set object
     m_object.set_key(m_key_buffer, strlen(m_key_buffer));
+    if (!m_compress_perc)
     m_object.add_value(m_value_buffer, new_size, true);
```
Comment (yael-lb) [inline obj_gen.cpp]: Indentation

File: memtier_benchmark.cpp (hunk 1 above)
Comment (tomerr-lb) [reply memtier_benchmark.cpp:202]: ok will do next time

File: obj_gen.cpp (hunk 2 above)
Comment (tomerr-lb) [reply obj_gen.cpp]: I've also had unnecessary cast of size . the casting is fixed now to match c++ although the C casting convention has the same functionality and c++ completely supports it

File: obj_gen.cpp (hunk 4 above)
Comment (tomerr-lb) [reply obj_gen.cpp]: oops thanks :)

**Comment by tomerr-lb** [comment]
retest please

**Pull request closed without merging**